src/libraries/System.DirectoryServices.Protocols/ref/System.DirectoryServices.Protocols.cs

```diff
@@ -372,7 +372,6 @@ public partial class LdapSessionOptions
         public bool RootDseCache { get { throw null; } set { } }
         public string SaslMethod { get { throw null; } set { } }
         public bool Sealing { get { throw null; } set { } }
-        [System.Runtime.Versioning.SupportedOSPlatformAttribute("windows")]
```

Comment:
Well by the design we should add an attribute for the newly supported platform, i.e: `[System.Runtime.Versioning.SupportedOSPlatformAttribute("linux")]` instead of removing the existing one.

Second, the change should be applied to the source too

Reply:
Updated by adding additional attribute as suggested

> Second, the change should be applied to the source too

Are we sure about this? This attribute is not applied to sources right now, this info is mainly used by analyzers that work off reference assemblies.

Reply:
> Are we sure about this? This attribute is not applied to sources right now, this info is mainly used by analyzers that work off reference assemblies.

Not sure if it is deleted, but I see it is added in the PR adding this to the ref https://github.com/dotnet/runtime/pull/52015/files

Reply:
It was removed in [this commit](https://github.com/dotnet/runtime/commit/32918374655b5e1c3fc7810bbb0c06a0d03265b7#diff-92620f7c0c09c1d3093c6951abe4fa4101cd82e9020d6563ff0a2196f2f8d558). 

What's the point of adding it to the sources? This attribute is only for use by analyzers which go off reference assemblies. The actual implementation is substituted via facade assembly mapping. Correct me if I'm wrong here but this metadata would not actually be used for anything if added to sources.

Reply:
> It was removed in this commit.

Yeah, i saw that, wonder why the API diff analyzer did not warn for that PR, and also in this PR CC @joperezr @safern 

> What's the point of adding it to the sources? This attribute is only for use by analyzers which go off reference assemblies. The actual implementation is substituted via facade assembly mapping. Correct me if I'm wrong here but this metadata would not actually be used for anything if added to sources.

In general, we keep ref and src in sync, at least if platform-specific API is called internally the src annotation could help, @joperezr @safern might say the better explanation for  having src and ref in sync

Reply:
@buyaa-n as @macsux explained, seems like back in 6.0 when we added support for this feature we just forgot to also remove the attribute from the ref, and I agree it would be good to check why APICompat didn't flag this. As for the fix, if the API is now supported everywhere, isn't the right design to just remove the constraint? as opposed to adding an attribute saying that the API is now supported? I was the one that suggested to just remove the current attribute from the ref so it would be good to understand the process for future reference

Reply:
> good to check why APICompat didn't flag this.

It is because we haven't updated API Compat to compare against 6.0.0 yet on main. It is still comparing against 5.0. I should update that shortly (we just realized that in the morning).

Reply:
@joperezr I don't think it would work on OSX in its current state as the necessary files are not included in a compilation for that OS. I *believe* OpenSSL libraries can run on OSX so theoretically the same Linux code can work there, but it would require additional code changes to make it work. Given that right now not all platforms are supported, it's premature to drop the attribute entirely.

Reply:
Not sure what you mean by "necessary files are not included" in OSX, but AFAIK it should work just fine there as well. OSX use the same implementation for the Property of SecureSocketLayer, as well as use the same InternalConnectToServer logic which is what reads this value and sets either ldap or ldaps appropriately, so I believe SSL is actually working for all of our three build flavors.

Reply:
The only difference between the Linux and OSX build that I'm aware of, is that the name of the native library that we PInvoke to in OSX is different than the one in Linux, but they are both Openldap and have the exact same surface area. If we have some behavior mismatch between OSX and Linux that is a bug, since the only reason why they are split is just that native library name.

Reply:
I'm not too familar with all MSBuild internal properties. The [`csproj` has conditional include](https://github.com/dotnet/runtime/blob/f9e0256ec1e88fae8ae53e3216d8403fb0070e77/src/libraries/System.DirectoryServices.Protocols/src/System.DirectoryServices.Protocols.csproj#L63-L74) for  `Condition="'$(TargetsUnix)' == 'true'"`. Is OSX falls under `TargetsUnix`? In that case, it probably will work, but I don't have an OSX environment to test on.

Reply:
> The csproj has conditional include for Condition="'$(TargetsUnix)' == 'true'". Is OSX falls under TargetsUnix? In that case, it probably will work, but I don't have an OSX environment to test on.

Yes, OSX falls under that category.

Reply:
I see, yeah I understand the confusion now. Basically you can think of it this way:

| Build Flavor | TargetsWindows | TargetsLinux | TargetsUnix | TargetsOSX |
| ------------- | ------------------ | ------------- | ------------- | ------------- |
| windows | true | false | false | false |
| linux | false | true | true | false |
| osx | false | false | true | true |